.github/workflows/docs.yml

```
@@ -0,0 +1,65 @@
+name: docs
+
+on:
+  push:
+    branches:
+      - add_documentation
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - uses: actions/setup-python@v4
+
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
+          python-version: 3.8
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+
+      - name: Conda info
+        shell: bash -l {0}
+        run: |
+          conda info
+          which python
+
+      - name: Mamba install FEniCS
+        shell: bash -l {0}
+        run: |
+          conda config --set always_yes yes
+          conda config --add channels conda-forge
+          mamba create -n fenicsproject -c conda-forge fenics 
+          conda activate fenicsproject
+          which python
+          python -c "from dolfin import *"
+
+
+      - name: Install dependencies
+        shell: bash -l {0}
+        run: |
+          conda activate fenicsproject
+          python -m pip install --upgrade pip
+          pip install --upgrade pip setuptools==65.7.0
+          pip install -r requirements.txt
+          pip install -r requirements-dev.txt
+
+      - name: Sphinx build
+        shell: bash -l {0}
+        run: |
+          conda activate fenicsproject
+          sphinx-build docs ./public
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/add_documentation' }}
```

```suggestion
        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
```
Using add_documentation branch is temp for testing (to publish the doc now)